File: c++/pointers/comparing_pointers.cpp
#include <iostream>

int main() {
  int numbers[] = {2, 5, 6, 7, 9};
  int *pi = numbers;
  int *pint = 0;

  if (pi == pint) {
    std::cout << "Both pointers point the same." << std::endl;
  }

  if (!pint) {
    std::cout << "Second pointer is not set on any place in the memory." << std::endl;
  }

  pint = numbers + 3;

  if (pint > pi && pi != 0) {
    std::cout << "First pointer is set on the element being nearest the beginning of the table than the second one." << std::endl;
  } else if (pint < pi && pint != 0) {
    std::cout << "First pointer is set on the element being nearest the end of the table than the second one." << std::endl;
  } else if (pint == pi != 0) {
    std::cout << "First pointer point the same element as the second one." << std::endl;
  }
}

File: pascal/pointers/comparing_pointers.pas
program ComparingPointers;

var
  numbers: array[0..5] of integer;
  pi: ^integer = @(numbers[0]);
  pint: ^integer = nil;

begin
  if (pi = pint) then
  begin
    writeln('Both pointers point the same.');
  end;

  if (pint = nil) then
  begin
    writeln('Second pointer is not set on any place in the memory.');
  end;

  pint := @(numbers[3]);

  if ((pint > pi) and (pi <> nil)) then
  begin
    writeln('First pointer is set on the element being nearest the beginning of the table than the second one.');
  end
  else if ((pint < pi) and (pint <> nil)) then
  begin
    writeln('First pointer is set on the element being nearest the end of the table than the second one.');
  end
  else if ((pint = pi) and (pi <> nil)) then
  begin
    writeln('First pointer point the same element as the second one.');
  end;
end.
